### PR TITLE
#1767 Add unit test to brokerApi

### PR DESCRIPTION
**Pull request by DavidMi-1**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1767)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by wu87988622** [comment]
retry manager-ut

**Comment by wu87988622** [comment]
@DavidMi-1 
Forgot to run Prettier?
You can use `yarn prepush` in local run unit test

**Comment by DavidMi-1** [comment]
> @DavidMi-1
> Forgot to run Prettier?
> You can use `yarn prepush` unit test in local

!! 明天, 今天到公司才能試

**Comment by konekoya** [comment]
Good first PR @DavidMi-1 👍 

A few things that you should know when opening a PR:

- A checklist will be added to your PR's description when a new PR is opened. Please read the checklist items and check them once a task is completed.
- As @wu87988622 have mentioned in the comment, you should run `yarn prepush` before opening the PR. This script will help you catch some linting/format issues and failing unit tests in your local so that you don't have to wait for the QA(CI) to run these checks and push fixes back and forth.

Happy testing :)

**Comment by konekoya** [comment]
@DavidMi-1 Could you also add this PR in the #1767? I've renamed the PR title :)

File: ohara-manager/client/src/api/__tests__/brokerApi.test.js [hunk 1]
```diff
@@ -185,3 +190,62 @@ describe('createBroker()', () => {
     expect(handleError).toHaveBeenCalledWith(res);
   });
 });
+
+describe('addNodeToBroker()', () => {
+  afterEach(jest.clearAllMocks);
```
Comment (konekoya) [inline ohara-manager/client/src/api/__tests__/brokerApi.test.js]: @DavidMi-1 Could you try moving this to the very top of this test file? See if it still works? That way we don't have to include it in every `describe` statement :)

File: ohara-manager/client/src/api/__tests__/brokerApi.test.js [hunk 2]
```diff
@@ -185,3 +190,62 @@ describe('createBroker()', () => {
...
+
+  const params = {
+    name: 'abc',
```
Comment (konekoya) [inline ohara-manager/client/src/api/__tests__/brokerApi.test.js]: Could you use our `generate` function here? It's inside the `/utils/generate.js` and it helps generate random/fake data for testing. We're adapting this pattern (using generated data) recently so I think it's a good opportunity to refactor here :)

**Comment by DavidMi-1** [comment]
> Good first PR @DavidMi-1 
> 
> A few things that you should know when opening a PR:
> 
> * A checklist will be added to your PR's description when a new PR is opened. Please read the checklist items and check them once a task is completed.
> * As @wu87988622 have mentioned in the comment, you should run `yarn prepush` before opening the PR. This script will help you catch some linting/format issues and failing unit tests in your local so that you don't have to wait for the QA(CI) to run these checks and push fixes back and forth as that's clearly time-consuming and tedious
> 
> Happy testing :)

ok, got it

File: ohara-manager/client/src/api/__tests__/brokerApi.test.js (hunk 1 above)
Comment (DavidMi-1) [reply ohara-manager/client/src/api/__tests__/brokerApi.test.js]: not sure it's still work or not. at least past test . XD

Comment (konekoya) [reply ohara-manager/client/src/api/__tests__/brokerApi.test.js]: If it's passing then I think it's working :)

File: ohara-manager/client/src/api/__tests__/brokerApi.test.js (hunk 2 above)
Comment (DavidMi-1) [reply ohara-manager/client/src/api/__tests__/brokerApi.test.js]: ok

File: ohara-manager/client/src/api/__tests__/brokerApi.test.js [hunk 3]
```diff
@@ -128,12 +131,10 @@ describe('fetchBrokers()', () => {
 });
 
 describe('createBroker()', () => {
-  afterEach(jest.clearAllMocks);
-
   const params = {
-    name: 'abc',
-    zookeeperClusterName: ['a', 'b'],
-    nodeNames: ['c', 'd'],
+    name: generate.name,
```
Comment (konekoya) [inline ohara-manager/client/src/api/__tests__/brokerApi.test.js]: You should invoke the `generate.name()` here since it's a function :)

Comment (DavidMi-1) [reply ohara-manager/client/src/api/__tests__/brokerApi.test.js]: got it, and console.log show ` generate.name() ` as a object content ` function(){(native code)}`

**Comment by konekoya** [comment]
@DavidMi-1 Looking good! Thanks for the contribution and congrats to the first PR 💯

**Pull request merged**